tests/unit/io-writer.spec.js

```diff
@@ -0,0 +1,138 @@
+import _ from 'lodash'
+import fs from 'fs'
+import test from 'tape'
+import Excel from 'exceljs'
+import tempWrite from 'temp-write'
+import Writer from '../../src/io/writer'
+
+const readXlsx = function readXlsx (filePath) {
+  const values = []
+  const workbook = new Excel.Workbook()
+  return workbook.xlsx.readFile(filePath)
+    .then(() => {
+      const worksheet = workbook.getWorksheet(1)
+      let headerLen = null
+      worksheet.eachRow((row) => {
+        let rowValues = row.values
+        rowValues.shift()
+
+        if (headerLen)
+          // older exceljs reader version trims last value
+          // if it is an empty string, so we will pad row
+          // to header length with empty string
+          rowValues = [
+            ...rowValues,
+            ...Array(headerLen - rowValues.length).fill(''),
+          ]
+        else
+          headerLen = rowValues.length
+
+        rowValues = _.map(rowValues,
+          item => (_.isObject(item) || _.isNil(item) ? '' : item))
+        values.push(rowValues)
+      })
+      return values
+    })
+}
+
+test(`Writer
+  should throw an error when outputFile is not specified`, (t) => {
+  const expectedRegexp = new RegExp('OutputFile was not specified', 'g')
+
+  const conf = {
+    exportFormat: 'csv',
+  }
+
+  t.throws(() => new Writer(conf), expectedRegexp,
+    'Should throw an error when creating writer object')
+  t.end()
+})
+
+test(`Writer
+  should throw an error when incorrect export format is specified`, (t) => {
+  const expectedRegexp = new RegExp('Unsupported file type: unknown,' +
+    ' supported formats are xlsx,csv', 'g')
+
+  const conf = {
+    outputFile: 'out.csv',
+    exportFormat: 'unknown',
+  }
+
+  t.throws(() => new Writer(conf), expectedRegexp,
+    'Should throw an error when creating writer object')
+  t.end()
+})
+
+test(`Writer
+  should throw an error when incorrect encoding is specified`, (t) => {
+  const expectedRegexp = new RegExp('Encoding does not exist: unknown', 'g')
+
+  const conf = {
+    outputFile: 'out.csv',
+    exportFormat: 'csv',
+    encoding: 'unknown',
+  }
+
+  t.throws(() => new Writer(conf), expectedRegexp,
+    'Should throw an error when creating writer object')
+  t.end()
+})
+
+test(`Writer
+  should return an error when writing fails`, (t) => {
+  const filePath = tempWrite.sync()
+  const expectedRegexp = new RegExp('must start with number,' +
+    ' buffer, array or string', 'g')
+
+  const writer = new Writer({
+    outputFile: filePath,
+    exportFormat: 'csv',
+  })
+
+  t.throws(() => writer._writeCsvRows(Buffer()), expectedRegexp,
```

Comment:
You are not checking if the writer fails, because executing `Buffer()` there returns a TypeError. ^^